.github/workflows/ci.yml

```diff
@@ -44,3 +44,50 @@ jobs:
 
     - name: Run Tests
       run: dotnet test test\OpenFeature.Tests\ --configuration Release --logger GitHubActions
+
+  packaging:
+    needs:
+    - unit-tests-linux
+    - unit-tests-windows
+
+    permissions:
+      contents: read
+      packages: write
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Setup .NET SDK
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: |
+          6.0.x
+          7.0.x
+
+    - name: Restore
+      run: dotnet restore
+
+    - name: Pack NuGet packages (CI versions)
+      if: startsWith(github.ref, 'refs/heads/')
+      run: dotnet pack --no-restore --version-suffix "ci.$(date -u +%Y%m%dT%H%M%S)+sha.${GITHUB_SHA:0:9}"
+
+    - name: Pack NuGet packages (PR versions)
+      if: startsWith(github.ref, 'refs/pull/')
+      run: dotnet pack --no-restore --version-suffix "pr.$(date -u +%Y%m%dT%H%M%S)+sha.${GITHUB_SHA:0:9}"
+
+    - name: Pack NuGet packages (Release versions)
+      if: startsWith(github.ref, 'refs/tags/v')
+      run: dotnet pack --no-restore
+
```

Comment:
I guess there's no harm in publishing proper release versions to GH as well, but we could remove this since we'll have one of these in nuget.

Reply:
Definitely don't want anyone mistaking this PR as proposing GH as an alternative to NuGet for official packages used in production code.

But I think the _packing_ step here isn't necessarily the problem (i.e. prefer as much consistency between release and non-release workflow runs as possible), but rather that the following step then _pushes_ the stable versions into the GH feed.

I'll update the conditions below to skip pushing in that case (and in the future we could have this push to NuGet, but not trying to shoehorn that in right now).

Reply:
> I'll update the conditions below to skip pushing in that case (and in the future we could have this push to NuGet, but not trying to shoehorn that in right now).

We push official releases to nuget with our release please flow when [PRs like this](https://github.com/open-feature/dotnet-sdk/pull/154) are merged. I'm not sure we'd want to publish anything to nuget other than that.

Reply:
Okay, yeah, seeing that now.

I think you're right that I'm creating more confusion by implying this _might_ do something for tag builds, so I'm just going to drop the notion of tag builds altogether in this PR.